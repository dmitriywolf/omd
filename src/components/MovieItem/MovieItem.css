.movie__item {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;

  width: 100%;
}

.movie__item-img {
  margin-bottom: auto;
}

.movie__item-content {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 15px 25px 10px;
}

.movie__item-content p {
  font-size: 16px;
}

.movie__item-title {
  margin: 10px 10px 0;
  font-size: 24px;
  text-align: center;
}

.movie__btn {
  border: 0;
  padding: 8px 15px;
  color: #ffffff;
  background-color: black;
}

.movie__btn:hover {
  cursor: pointer;
  opacity: 0.8;
}
