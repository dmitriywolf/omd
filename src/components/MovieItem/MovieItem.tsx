import React from 'react';
import {Link} from "react-router-dom";
import './MovieItem.css';

type CompProp = {
  Title: string,
  Year: number,
  Poster: string,
  imdbID: string
}

const MovieItem: React.FC<CompProp> = ({Title, Year, Poster, imdbID}) => {
  return (
    <div className="card movie__item">
      <div className="card-image movie__item-img">
        <img src={`${Poster}`} alt={`${Title}`}/>
      </div>
      <h3 className="movie__item-title">{Title}</h3>
      <div className="movie__item-content">
        <p> {`Year: ${Year}`}</p>

        <Link to={`/movies/${imdbID}`}>
          <button className="movie__btn">Open</button>
        </Link>
      </div>
    </div>
  )
};

export default MovieItem;
