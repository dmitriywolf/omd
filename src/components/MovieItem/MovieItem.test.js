import React from 'react';
import MovieItem from './MovieItem';
import {shallow} from 'enzyme';
import toJson from "enzyme-to-json";


it('should render MovieItem component with props', () => {
  const component = shallow(<MovieItem imdbID="tt34444" Poster="src/img.png" Title="Title movie" Year="2008"/>);
  expect(toJson(component)).toMatchSnapshot();
});

it('should render MovieItem component without props', () => {
  const component = shallow(<MovieItem/>);
  expect(toJson(component)).toMatchSnapshot();
});

