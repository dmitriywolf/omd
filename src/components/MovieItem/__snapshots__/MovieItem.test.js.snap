// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieItem component with props 1`] = `
<div
  className="card movie__item"
>
  <div
    className="card-image movie__item-img"
  >
    <img
      alt="Title movie"
      src="src/img.png"
    />
  </div>
  <h3
    className="movie__item-title"
  >
    Title movie
  </h3>
  <div
    className="movie__item-content"
  >
    <p>
       
      Year: 2008
    </p>
    <Link
      to="/movies/tt34444"
    >
      <button
        className="movie__btn"
      >
        Open
      </button>
    </Link>
  </div>
</div>
`;

exports[`should render MovieItem component without props 1`] = `
<div
  className="card movie__item"
>
  <div
    className="card-image movie__item-img"
  >
    <img
      alt="undefined"
      src="undefined"
    />
  </div>
  <h3
    className="movie__item-title"
  />
  <div
    className="movie__item-content"
  >
    <p>
       
      Year: undefined
    </p>
    <Link
      to="/movies/undefined"
    >
      <button
        className="movie__btn"
      >
        Open
      </button>
    </Link>
  </div>
</div>
`;
